### PR TITLE
[SPARK-15394][ML][DOCS] User guide typos and grammar audit

### PR DESCRIPTION
**Pull request by sethah**
## What changes were proposed in this pull request?

Correct some typos and incorrectly worded sentences.
## How was this patch tested?

Doc changes only.

Note that many of these changes were identified by @whomfire01

### CODE REVIEW

**Comment by sethah** [comment]
cc @holdenk Could you take a look?

File: docs/ml-collaborative-filtering.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ best parameter learned from a sampled subset to the full dataset and expect simi
 <div class="codetabs">
 <div data-lang="scala" markdown="1">
 
-In the following example, we load rating data from the
+In the following example, we load ratings data from the
```
Comment (srowen) [inline docs/ml-collaborative-filtering.md]: Although I think "rating data" is actually OK, so is "ratings data". I think the whole change LGTM

**Comment by holdenk** [comment]
LGTM as well :)

**Comment by mengxr** [comment]
Merged into master and branch-2.0. Thanks!

**Pull request closed without merging**